synapse/events/__init__.py

```diff
@@ -408,7 +407,7 @@ def items(self) -> List[Tuple[str, Optional[Any]]]:
     def keys(self) -> Iterable[str]:
         return self._dict.keys()
 
-    def prev_event_ids(self) -> Sequence[str]:
+    def prev_event_ids(self) -> List[str]:
```

Comment:
Since we're returning a brand-new list with no references to any internal data I don't think there's a huge reason to return something immutable. My bias would be to leave it as `List[str]` for now, but that's only a weak preference.